pkg/space/space_test.go

```diff
@@ -53,3 +54,34 @@ func TestNewSubSpace(t *testing.T) {
 	)
 	assert.Equal(t, expectedSubSpace, subSpace)
 }
+
+func TestNewSubSubSpace(t *testing.T) {
+	// given
+	srClusterRoles := []string{commoncluster.RoleLabel(commoncluster.Tenant)}
+	sr := spacerequesttest.NewSpaceRequest("jane", "jane-tenant",
+		spacerequesttest.WithTierName("appstudio"),
+		spacerequesttest.WithTargetClusterRoles(srClusterRoles))
+	sr2 := spacerequesttest.NewSpaceRequest("jane", "jane-tenant",
+		spacerequesttest.WithTierName("appstudio"),
+		spacerequesttest.WithTargetClusterRoles(srClusterRoles))
+	parentSpace := spacetest.NewSpace(test.HostOperatorNs, "parentSpace")
+	subSpace := NewSubSpace(sr, parentSpace)
+
+	// when
+	subSubSpace := NewSubSpace(sr2, subSpace)
+
+	// then
+	expectedSubSubSpace := spacetest.NewSpace(test.HostOperatorNs, fmt.Sprintf("%s-%s", parentSpace.Name, sr2.UID[:5]),
+		spacetest.WithSpecParentSpace(subSpace.GetName()),
+		spacetest.WithTierName("appstudio"),
+		spacetest.WithSpecTargetClusterRoles([]string{cluster.RoleLabel(cluster.Tenant)}),
+		spacetest.WithLabel(toolchainv1alpha1.SpaceRequestLabelKey, sr.GetName()),
+		spacetest.WithLabel(toolchainv1alpha1.SpaceRequestNamespaceLabelKey, sr.GetNamespace()),
+		spacetest.WithLabel(toolchainv1alpha1.ParentSpaceLabelKey, subSpace.GetName()),
+	)
+	assert.Equal(t, expectedSubSubSpace, subSubSpace)
+
+	// also assert that names don't grow in length as we increase nesting
+	assert.Equal(t, len(subSpace.Name), len(subSubSpace.Name))
+	assert.NotEqual(t, subSpace.Name, subSubSpace.Name)
+}
```

Comment:
We should rename sr and expect that as name and namespace in the labels:
```suggestion
func TestNewSubSubSpace(t *testing.T) {
	// given
	srClusterRoles := []string{commoncluster.RoleLabel(commoncluster.Tenant)}
	sr := spacerequesttest.NewSpaceRequest("jane", "jane-tenant",
		spacerequesttest.WithTierName("appstudio"),
		spacerequesttest.WithTargetClusterRoles(srClusterRoles))
	parentSpace := spacetest.NewSpace(test.HostOperatorNs, "parentSpace")
	subSpace := NewSubSpace(sr, parentSpace)
	// create another spacerequest in the subSpace namespace
	sr2 := spacerequesttest.NewSpaceRequest("jane2", subSpace.GetName()+"-tenant",
		spacerequesttest.WithTierName("appstudio"),
		spacerequesttest.WithTargetClusterRoles(srClusterRoles))

	// when
	subSubSpace := NewSubSpace(sr2, subSpace)

	// then
	expectedSubSubSpace := spacetest.NewSpace(test.HostOperatorNs, fmt.Sprintf("%s-%s", parentSpace.Name, sr2.UID[:5]),
		spacetest.WithSpecParentSpace(subSpace.GetName()),
		spacetest.WithTierName("appstudio"),
		spacetest.WithSpecTargetClusterRoles([]string{cluster.RoleLabel(cluster.Tenant)}),
		spacetest.WithLabel(toolchainv1alpha1.SpaceRequestLabelKey, sr2.GetName()),
		spacetest.WithLabel(toolchainv1alpha1.SpaceRequestNamespaceLabelKey, sr2.GetNamespace()),
		spacetest.WithLabel(toolchainv1alpha1.ParentSpaceLabelKey, subSpace.GetName()),
	)
	assert.Equal(t, expectedSubSubSpace, subSubSpace)

	// also assert that names don't grow in length as we increase nesting
	assert.Equal(t, len(subSpace.Name), len(subSubSpace.Name))
	assert.NotEqual(t, subSpace.Name, subSubSpace.Name)
}
```

Reply:
Fixed in bec33a7.